packages/browser/src/autocapture-utils.ts

```diff
@@ -338,56 +341,35 @@ export function shouldCaptureDomEvent(
 }
 
 /*
- * Check whether a DOM element should be "captured" or if it may contain sensitive data
- * using a variety of heuristics.
+ * Check whether an element or its parents have been explicitly marked to not capture.
+ * Looks for 'ph-sensitive' or 'ph-no-capture' classes.
  * @param {Element} el - element to check
- * @returns {boolean} whether the element should be captured
+ * @returns {boolean} whether the element is marked as no-capture
  */
-export function shouldCaptureElement(el: Element): boolean {
+export function isExplicitNoCapture(el: Element): boolean {
     for (let curEl = el; curEl.parentNode && !isTag(curEl, 'body'); curEl = curEl.parentNode as Element) {
         const classes = getClassNames(curEl)
         if (includes(classes, 'ph-sensitive') || includes(classes, 'ph-no-capture')) {
-            return false
-        }
-    }
-
-    if (includes(getClassNames(el), 'ph-include')) {
-        return true
-    }
-
-    // don't include hidden or password fields
-    const type = (el as HTMLInputElement).type || ''
-    if (isString(type)) {
-        // it's possible for el.type to be a DOM element if el is a form with a child input[name="type"]
-        switch (type.toLowerCase()) {
-            case 'hidden':
-                return false
-            case 'password':
-                return false
-        }
-    }
-
-    // filter out data from fields that look like sensitive fields
-    const name = (el as HTMLInputElement).name || el.id || ''
-    // See https://github.com/posthog/posthog-js/issues/165
-    // Under specific circumstances a bug caused .replace to be called on a DOM element
-    // instead of a string, removing the element from the page. Ensure this issue is mitigated.
-    if (isString(name)) {
-        // it's possible for el.name or el.id to be a DOM element if el is a form with a child input[name="name"]
-        const sensitiveNameRegex =
-            /^cc|cardnum|ccnum|creditcard|csc|cvc|cvv|exp|pass|pwd|routing|seccode|securitycode|securitynum|socialsec|socsec|ssn/i
-        if (sensitiveNameRegex.test(name.replace(/[^a-zA-Z0-9]/g, ''))) {
-            return false
```

Comment:
we have a more broader scope (i.e. safer) check for all of this in `isSensitiveElement`